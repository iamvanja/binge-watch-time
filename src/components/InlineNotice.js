import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

const InlineNotice = ({ children, active, type, size }) => {
  if (!children || !active) {
    return null
  }
  const componentClasses = classnames(
    'callout', {
      [type]: !!type,
      [size]: !!size
    }
  )

  return (
    <div className={componentClasses}>
      {children}
    </div>
  )
}

InlineNotice.defaultProps = {
  active: true,
  size: 'normal'
}

InlineNotice.propTypes = {
  children: PropTypes.node,
  active: PropTypes.bool,
  type: PropTypes.oneOf([
    'warning',
    'alert',
    'success',
    'primary',
    'secondary'
  ]),
  size: PropTypes.oneOf([
    'small',
    'normal',
    'large'
  ])
}

export default InlineNotice
