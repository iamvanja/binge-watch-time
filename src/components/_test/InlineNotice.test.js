import React from 'react'
import { shallow } from 'enzyme'
import InlineNotice from '../InlineNotice'

describe('InlineNotice', () => {
  let props
  let wrapper
  let children
  const component = () => {
    if (!wrapper) {
      wrapper = shallow(
        <InlineNotice {...props}>{children}</InlineNotice>
      )
    }
    return wrapper
  }

  beforeEach(() => {
    props = {
      type: undefined,
      size: undefined,
      active: undefined
    }
    wrapper = undefined
    children = 'TEST'
  })

  it('renders an InlineNotice', () => {
    expect(component()).toMatchSnapshot()
  })

  it('renders a primary InlineNotice', () => {
    props.type = 'primary'
    expect(component()).toMatchSnapshot()
  })

  it('renders a small InlineNotice', () => {
    props.size = 'small'
    expect(component()).toMatchSnapshot()
  })

  it('does not render when no children passed', () => {
    children = null
    expect(component().html()).toEqual(null)
  })

  it('does not render when inactive', () => {
    props.active = false
    expect(component().html()).toEqual(null)
  })
})
