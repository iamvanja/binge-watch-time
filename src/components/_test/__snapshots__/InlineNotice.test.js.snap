// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineNotice renders a primary InlineNotice 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InlineNotice
    active={true}
    size="normal"
    type="primary"
>
    TEST
</InlineNotice>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "TEST",
      "className": "callout primary normal",
    },
    "ref": null,
    "rendered": "TEST",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "TEST",
        "className": "callout primary normal",
      },
      "ref": null,
      "rendered": "TEST",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`InlineNotice renders a small InlineNotice 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InlineNotice
    active={true}
    size="small"
    type={undefined}
>
    TEST
</InlineNotice>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "TEST",
      "className": "callout small",
    },
    "ref": null,
    "rendered": "TEST",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "TEST",
        "className": "callout small",
      },
      "ref": null,
      "rendered": "TEST",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`InlineNotice renders an InlineNotice 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InlineNotice
    active={true}
    size="normal"
    type={undefined}
>
    TEST
</InlineNotice>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "TEST",
      "className": "callout normal",
    },
    "ref": null,
    "rendered": "TEST",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "TEST",
        "className": "callout normal",
      },
      "ref": null,
      "rendered": "TEST",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
